src/main/scala/mesosphere/marathon/core/launcher/impl/InstanceOpFactoryImpl.scala

```diff
@@ -275,12 +274,19 @@ object InstanceOpFactoryImpl {
       state = InstanceState(InstanceStatus.Created, since, pod.version, healthy = None),
       tasksMap = taskIDs.map { id =>
 
+        // the task level host ports are needed for fine-grained status/reporting later on
+        val taskHostPorts: Seq[Int] = id.containerName.map { ctName =>
+          requestedPortsPerCT.zip(hostPorts).collect {
+            case ((name, Some(_)), Some(allocatedPort)) if name == ctName => allocatedPort
+          }
+        }.getOrElse(Seq.empty[Int])
+
```

Comment:
If you do this computation within the `.map` it will be evaluated n times, where n is the number of containers. Please move that out of the map.